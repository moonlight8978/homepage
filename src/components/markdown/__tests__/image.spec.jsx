import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Image from '../image'

test('it render the image correctly', () => {
  const { container, getByText, getByTestId } = render(
    <Image src="/sample.jpg" alt="Sample image" />
  )

  expect(getByText('Sample image')).toBeInTheDocument()

  const image = getByTestId('markdown-image')
  expect(image.getAttribute('src')).toBe('/sample.jpg')

  expect(container.firstChild).toMatchSnapshot()
})
