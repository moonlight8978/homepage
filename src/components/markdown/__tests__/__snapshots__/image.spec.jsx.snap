// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it render the image correctly 1`] = `
<div
  class="container"
>
  <img
    alt="Sample image"
    class="image"
    data-testid="markdown-image"
    src="/sample.jpg"
  />
  <div
    class="caption"
  >
    Sample image
  </div>
</div>
`;
